test/test_utils/test_common.py

```diff
@@ -0,0 +1,31 @@
+"""
+This tests the functionality in autoPyTorch/utils/common.
+"""
+import pytest
+
+from autoPyTorch.utils.common import autoPyTorchEnum
+
+
+class SubEnum(autoPyTorchEnum):
+    x = "x"
+    y = "y"
+
+
+@pytest.mark.parametrize('iter',
+                         [[SubEnum.x],
+                          ["x"],
+                          {SubEnum.x: "hello"},
+                          {'x': 'hello'}])
```

Comment:
I made it immutable just in case and stronger.
```suggestion
                         ([SubEnum.x],
                          ["x"],
                          {SubEnum.x: "hello"},
                          {'x': 'hello'},
                          SubEnum,
                          ["x", "y"]))
```